apps/web/src/components/AssessmentStepTracker/AssessmentStepTracker.tsx

```diff
@@ -4,40 +4,54 @@ import { useIntl } from "react-intl";
 import { Board } from "@gc-digital-talent/ui";
 import { getLocalizedName } from "@gc-digital-talent/i18n";
 import { unpackMaybes } from "@gc-digital-talent/helpers";
+import { Pool } from "@gc-digital-talent/graphql";
 
-import { Pool } from "~/api/generated";
 import applicationMessages from "~/messages/applicationMessages";
 
 import ResultsDetails from "./ResultsDetails";
 import AssessmentResults from "./AssessmentResults";
-import { groupPoolCandidatesByStep } from "./utils";
+import {
+  ResultFilters,
+  filterResults,
+  groupPoolCandidatesByStep,
+  defaultFilters,
+} from "./utils";
+import Filters from "./Filters";
 
 export interface AssessmentStepTrackerProps {
   pool: Pool;
 }
 
 const AssessmentStepTracker = ({ pool }: AssessmentStepTrackerProps) => {
   const intl = useIntl();
+  const [filters, setFilters] = React.useState<ResultFilters>(defaultFilters);
   const steps = unpackMaybes(pool.assessmentSteps);
   const candidates = unpackMaybes(pool.poolCandidates);
   const groupedSteps = groupPoolCandidatesByStep(steps, candidates);
 
+  const filteredSteps = React.useMemo(() => {
+    return filterResults(filters, groupedSteps);
+  }, [groupedSteps, filters]);
```

Comment:
Is it worth memoizing this if groupedSteps isn't also memoized?

Reply:
Yeah, that makes sense. Removed in [remove memoization](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9158/commits/903299c8eb91350d66ab89acef72b7415da7b512)